crates/engine/src/api/operation.rs

```diff
@@ -1,6 +1,7 @@
 use crate::EngineError;
 
 /// Storage operations
+/// TODO: refactor this trait, require `&mut self` for write operations
```

Comment:
Please leave an issue to track.

Reply:
I think this TODO could be removed since rocksdb does not requires `&mut self`.